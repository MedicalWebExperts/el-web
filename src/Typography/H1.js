import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  text: PropTypes.string.isRequired,
};

const H1 = ({ text }) => (
  <h1>{text}</h1>
);

H1.propTypes = propTypes;

export default H1;
