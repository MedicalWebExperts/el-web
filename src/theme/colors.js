export default {
  default: '#000000',
  primary: '#0000ff',
  secondary: '#00b33c',
};
