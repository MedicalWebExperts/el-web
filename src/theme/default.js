export default {
  colors: {
    primary: '#0000ff',
    secondary: '#00b33c',
    default: '#000000',
  },
  font: {
    primary: 'Georgia, serif',
    secondary: 'sans-serif',
  },
};
