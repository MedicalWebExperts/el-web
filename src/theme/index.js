import { withTheme } from 'theming';
import ThemeProvider from './provider';
import DefaultTheme from './default';

module.exports = {
  ThemeProvider,
  withTheme,
  DefaultTheme,
};
