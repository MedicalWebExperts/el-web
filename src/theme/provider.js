import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as ThemeProviderBase } from 'theming';
import DefaultTheme from './default';

const propTypes = {
  children: PropTypes.element.isRequired,
  theme: PropTypes.oneOfType([PropTypes.shape({})]),
};

const defaultProps = {
  theme: {},
};

const ThemeProvider = ({ children, theme }) => {
  // check a util funcion for deep clone/merge
  const mergedTheme = Object.keys(theme).length === 0 ? DefaultTheme : theme;

  return (
    <ThemeProviderBase theme={mergedTheme}>
      {children}
    </ThemeProviderBase>
  );
};

ThemeProvider.propTypes = propTypes;
ThemeProvider.defaultProps = defaultProps;

export default ThemeProvider;
