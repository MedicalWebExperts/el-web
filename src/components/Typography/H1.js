import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from '../../theme';

const propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
  theme: PropTypes.shape({}).isRequired,
};

const defaultProps = {
  color: 'default',
};

const H1 = ({ text, theme, color }) => (
  <h1>
    <style jsx>{`
      h1 {
        color: ${theme.colors[color]}
      }
    `}
    </style>
    {text}
  </h1>
);

H1.propTypes = propTypes;
H1.defaultProps = defaultProps;

export default withTheme(H1);
