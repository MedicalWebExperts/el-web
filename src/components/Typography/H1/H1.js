import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from '../../../theme';

const propTypes = {
  text: PropTypes.string.isRequired,
  secondary: PropTypes.bool,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
  theme: PropTypes.shape({}).isRequired,
};

const defaultProps = {
  color: 'default',
  secondary: false,
};

const H1 = ({
  text,
  theme,
  color,
  secondary,
}) => (
  <h1>
    <style jsx>{`
      h1 {
        color: ${theme.colors[color]}
        font-family: ${secondary ? theme.font.secondary : theme.font.primary}
      }
    `}
    </style>
    {text}
  </h1>
);

H1.propTypes = propTypes;
H1.defaultProps = defaultProps;

export default withTheme(H1);
