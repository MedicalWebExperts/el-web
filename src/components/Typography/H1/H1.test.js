import React from 'react';
import renderer from 'react-test-renderer';

import H1 from './H1';
import { ThemeProvider } from '../../../theme';

describe('Component H1 Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer
      .create(<ThemeProvider><H1 text="Heading 1" /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><H1 text="Heading 1" color="primary" /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><H1 text="Heading 1" secondary /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
