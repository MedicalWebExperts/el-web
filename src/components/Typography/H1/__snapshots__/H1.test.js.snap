// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component H1 Snapshot primary color of the theme 1`] = `
<h1
  className="jsx-2310562709"
>
  Heading 1
</h1>
`;

exports[`Component H1 Snapshot secondary font-family of the theme 1`] = `
<h1
  className="jsx-2537203115"
>
  Heading 1
</h1>
`;

exports[`Component H1 Snapshot without arguments, take the default values 1`] = `
<h1
  className="jsx-709745237"
>
  Heading 1
</h1>
`;
